pkg/util/pki/csr_test.go

```diff
@@ -553,3 +556,42 @@ func Test_buildKeyUsagesExtensionsForCertificate(t *testing.T) {
 		})
 	}
 }
+
+func TestIPAddressesForCertificate(t *testing.T) {
+	tests := map[string]struct {
+		crt  *v1.Certificate
+		want []net.IP
+	}{
+		"valid ips should be returned": {
+			// Cannot use gen.Certificate because "gen" imports "pki".
+			crt: &v1.Certificate{Spec: v1.CertificateSpec{
+				IPAddresses: []string{
+					"127.0.0.1",
+					"127.0.0.2",
+				},
+			}},
+			want: []net.IP{
+				net.ParseIP("127.0.0.1"),
+				net.ParseIP("127.0.0.2"),
+			},
+		},
+		"a malformed ip is silently skipped": {
```

Comment:
This sounds scarier than it is. A Certificate cannot be created with a malformed IP address as this is checked at admission time.

Reply:
Makes sense... but I might end up using `IPAddressesForCertificate` and not know that we have such assumptions in the "higher layers".

I think it makes sense to talk about "silently skipped" since that's exactly what we do: we skip IPs that are malformed in the `net.ParseIP` sense.